yarn-project/aztec-nr/aztec/src/history/contract_inclusion.nr

```diff
@@ -0,0 +1,47 @@
+use dep::protocol_types::{
+    abis::new_contract_data::NewContractData as ContractLeafPreimage,
+    address::{AztecAddress, EthAddress},
+};
+use dep::std::merkle::compute_merkle_root;
+
+use crate::{
+    context::PrivateContext,
+    oracle::get_membership_witness::get_contract_membership_witness,
+};
+
+// Proves that the contract address, portal address and function tree root form a valid contract preimage of a leaf
+// which exists at block `block_number`.
+// Note: This can be used to approximate a factory pattern --> a factory contract could perform this proof and that
```

Comment:
Would be useful to add a ref to https://github.com/AztecProtocol/aztec-packages/issues/3695 to ensure that it is not left as it hence it is pretty vulnerable atm.

Reply:
Addressed the vulnerability in newer commits so no longer relevant.